src/inet/IPAddress.h

```diff
@@ -201,6 +201,9 @@ class DLL_EXPORT IPAddress
      *  address in network byte order.
      */
     uint32_t Addr[4];
+#if CHIP_SYSTEM_CONFIG_USE_LWIP && LWIP_IPV6_SCOPES
+    uint8_t Zone;
```

Comment:
should this default to `= IP6_NO_ZONE` ? Or are we always setting it to something? 

Generally how about other constructors:
  - The default constructor
  - IPV4 constructor
  - the otIp6Address (is that even a thing? would it be supported with LWIP and scopes enabled?)

Reply:
I think `Zone` should be set to something whenever `Addr` is set to something. Defaulting to IP6_NO_ZONE won't hurt though.

IPv4 doesn't need zones and I think OT has nothing to do with LWIP routing. Correct me if I am wrong.